### PR TITLE
Add check/test jobs to cluster-api-provider-openstack

### PR DESCRIPTION
**Pull request by flaper87**
Fixes https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/41

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cluster-api-provider-openstack/cluster-api-provider-openstack-presubmits.yaml [hunk 1]
```diff
@@ -23,3 +23,35 @@ presubmits:
         resources:
           requests:
             memory: "6Gi"
+  - name: pull-cluster-api-provider-openstack-test
```
Comment (krzyzacy) [inline config/jobs/kubernetes-sigs/cluster-api-provider-openstack/cluster-api-provider-openstack-presubmits.yaml]: you should utilize podutil to make new jobs now :-)

(aka add `decorate: true`, and you can save all bootstrap args, for example: https://github.com/kubernetes/test-infra/blob/d6f6b8a1abcacb348188cd2753aec0ab56954ad5/config/jobs/kubernetes/org/kubernetes-org-jobs.yaml#L14-L21)

Comment (krzyzacy) [reply config/jobs/kubernetes-sigs/cluster-api-provider-openstack/cluster-api-provider-openstack-presubmits.yaml]: (looks like we'll need to update the sample config in readme)

Comment (flaper87) [reply config/jobs/kubernetes-sigs/cluster-api-provider-openstack/cluster-api-provider-openstack-presubmits.yaml]: Thanks for the review! Will update accordingly :)

File: testgrid/config.yaml [hunk 2]
```diff
@@ -2453,6 +2453,12 @@ test_groups:
 - name: pull-cluster-api-provider-openstack-make
   gcs_prefix: kubernetes-jenkins/pr-logs/directory/pull-cluster-api-provider-openstack-make
   num_columns_recent: 20
+- name: pull-cluster-api-provider-openstack-test
+  gcs_prefix: kubernetes-jenkins/pr-logs/directory/pull-cluster-api-provider-openstack-test
+  num_columns_recent: 20
+- name: pull-cluster-api-provider-openstack-check
+  gcs_prefix: kubernetes-jenkins/pr-logs/directory/pull-cluster-api-provider-openstack-check
+  num_columns_recent: 20
```
Comment (krzyzacy) [inline testgrid/config.yaml]: you need to put the testgroup in one of the dashboard, probably https://github.com/kubernetes/test-infra/blob/3213b23d52905dca94623adcd41c7e24dbf30933/testgrid/config.yaml#L5804-L5807

File: config/jobs/kubernetes-sigs/cluster-api-provider-openstack/OWNERS [hunk 3]
```diff
@@ -0,0 +1,4 @@
+approvers:
+- flaper87
+- Lion-Wei
+- chaoscaffe
```
Comment (m1093782566) [inline config/jobs/kubernetes-sigs/cluster-api-provider-openstack/OWNERS]: @flaper87 You can put my github handle here :)

File: testgrid/config.yaml (hunk 2 above)
Comment (flaper87) [reply testgrid/config.yaml]: thanks again for your review! :+1:

File: config/jobs/kubernetes-sigs/cluster-api-provider-openstack/OWNERS (hunk 3 above)
Comment (flaper87) [reply config/jobs/kubernetes-sigs/cluster-api-provider-openstack/OWNERS]: done

Comment (m1093782566) [reply config/jobs/kubernetes-sigs/cluster-api-provider-openstack/OWNERS]: cool!

**Comment by Lion-Wei** [comment]
@flaper87 Hi, seems like the ci job have some error. 
xref: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/36#issuecomment-428803166

error message: 
```
find: `/home/prow/go/src/sigs.k8s.io/cluster-api-provider-openstack': No such file or directory
Makefile:33: *** Please run 'make' from /home/prow/go/src/sigs.k8s.io/cluster-api-provider-openstack. Current directory is /home/prow/go/src/github.com/kubernetes-sigs/cluster-api-provider-openstack.  Stop.
```

**Pull request merged**